packages/manager/apps/catalog/src/components/Filters/Filters.tsx

```diff
@@ -31,12 +32,28 @@ const Filters: React.FC<FiltersProps> = ({
   const [selectedUniverses, setSelectedUniverses] = useState<string[]>([]);
   const [hasInteracted, setHasInteracted] = useState<boolean>(false);
 
+  const location = useLocation();
+
   const setFilters = () => {
     setParentSelectedCategories(selectedCategories);
     setParentSelectedUniverses(selectedUniverses);
     onApply();
   };
 
+  useEffect(() => {
+    const params = new URLSearchParams(location.search);
+    const universesURL = params.get('universes');
+    const categoriesURL = params.get('categories');
+    if (categoriesURL) {
+      setSelectedCategories(categoriesURL.split(','));
+      setParentSelectedCategories(categoriesURL.split(','));
+    }
+    if (universesURL) {
+      setSelectedUniverses(universesURL.split(','));
+      setParentSelectedUniverses(universesURL.split(','));
+    }
```

Comment:
I feel that we should put this in a function and the content of the useEffect of the index.tsx file because, if for any reason, we change the parameter or we add new parameters, it will be easier to refactor if the functions are next to each other.
They will be easy to unit test, also.

You can call it `getFilterParamsFromUrl` and return something like `{universes: string[]' categories: string[]}`

Reply:
Done, getFilterParamsFromUrl created ✅

Reply:
Ok nice
Can you move it to the same file as the other function that create url from params ?